src/overlay/src/Overlay.js

```diff
@@ -162,6 +162,10 @@ class Overlay extends React.Component {
     }
   }
 
+  componentDidMount() {
+    document.body.style.overflow = 'hidden'
```

Comment:
Shouldn't this be applied only when `this.props.isShown`?

Reply:
@mshwery I am not sure about that.

Reply:
Yeah I think we need to monitor the prop from changing. Maybe we can use `onEnter` and `onExit` event instead.

```jsx
<Transition
  appear
  unmountOnExit
  timeout={ANIMATION_DURATION}
  in={isShown && !exiting}
  onExit={onExit} // onExit={this.handleExit}
  onExiting={this.handleExiting}
  onExited={this.handleExited}
  onEnter={onEnter} // => onEnter={this.handleEnter}
  onEntering={this.handleEntering}
  onEntered={this.handleEntered}
>
```

Then in the `handleEnter/Exit` method we can do the following:

```js
handleEnter = () => {
  this.previousOverflow = document.body.style.overflow
  document.body.style.overflow = 'hidden'
  this.props.onEnter()
}

handleExit = () => {
  document.body.style.overflow = this.previousOverflow
  this.prop.onExit() 
}
```

I think the above should work. @mshwery do you seen any issues here?

Reply:
Would be nice if this can be moved forward, as I really need this feature.

Reply:
I use dialog to show content and usually it has its own scrollbar. Without this change, when users reach the end of the dialog and keep scrolling, the page will scroll too. Something like this https://codesandbox.io/s/18lprv8rr3